src/main/java/wolox/training/services/OpenLibrary.java

```diff
@@ -24,55 +29,94 @@
     @Autowired
     private Environment env;
 
+    @Autowired
+    private BookRepository bookRepository;
+
     public OpenLibrary(RestTemplateBuilder restTemplateBuilder) {
         restTemplate = restTemplateBuilder.build();
     }
 
-    private JsonElement doIsbnRequest(String isbn) throws Exception {
+    private String doIsbnRequest(String isbn) throws IOException {
         String baseUrl = env.getProperty("services.openLibraryBooksBaseUrl");
         String url = baseUrl + "/books?bibkeys=ISBN:" + isbn + "&format=json&jscmd=data";
         URL urlPath = new URL(url);
         HttpURLConnection con =(HttpURLConnection) urlPath.openConnection();
         con.setRequestMethod("GET");
-        if (con.getResponseCode() != HttpURLConnection.HTTP_OK)
-            return null;
+        int code = con.getResponseCode();
+        if (code == HttpURLConnection.HTTP_NOT_FOUND) {
+            throw new EntityNotFoundException(Book.class);
+        }
+        if (code != HttpURLConnection.HTTP_OK) {
+            throw new IOException();
+        }
         BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
         String inputLine;
         StringBuffer response = new StringBuffer();
-        while ((inputLine = in.readLine()) != null)
+        while ((inputLine = in.readLine()) != null) {
             response.append(inputLine);
+        }
         in.close();
 
-        return JsonParser.parseString(response.toString()).getAsJsonObject();
+        return response.toString();
     }
 
+    public static OpenLibraryBookDTO buildDTOFromJsonString(String isbn, String jsonString){
```

Comment:
This method could be private, the only method that uses it belongs to the same class

Reply:
Whoops.. i actually use this method when implementing Wiremock to test so that's why it's public

Reply:
Ah! It's ok then! :D